consumer.go

```diff
@@ -401,10 +401,22 @@ func (child *partitionConsumer) HighWaterMarkOffset() int64 {
 }
 
 func (child *partitionConsumer) responseFeeder() {
+feederLoop:
 	for response := range child.feeder {
-		switch err := child.handleResponse(response); err {
+		switch msgs, err := child.parseResponse(response); err {
 		case nil:
-			break
+			for i, msg := range msgs {
+				select {
+				case child.messages <- msg:
+				case <-time.After(child.conf.Consumer.MaxProcessingTime):
+					child.broker.done <- child
+					for _, msg = range msgs[i:] {
+						child.messages <- msg
+					}
+					child.broker.input <- child
+					continue feederLoop
```

Comment:
This is happening in its own goroutine. If the timeout triggers, we send a message back to the brokerConsumer telling it to take us out of the pool, and then we feed the rest of the messages to the user at our leisure (so yes, the second message send probably blocks until the user catches up, but it doesn't block any *other* partitions).

Once we've flushed all our messages, we put ourselves back in the pool